internal/infrastructure/manager.go

```diff
@@ -0,0 +1,43 @@
+package infrastructure
+
+import (
+	"errors"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	clicfg "sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/infrastructure/kubernetes"
+)
+
+// Manager provides the scaffolding for managing infrastructure.
+type Manager struct {
+	Kubernetes *kubernetes.Infra
+}
```

Comment:
Do we expect there to multiple things here? Before I got to this file, I was assuming `Manager` was going to be an interface that `kubernetes.Infra` implements.

Reply:
Yes, the struct will contain a field for every supported provider. I created https://github.com/envoyproxy/gateway/issues/182 to track the need to move this to an interface.